packages/key4hep-stack/package.py

```diff
@@ -83,7 +83,7 @@ class Key4hepStack(BundlePackage, Key4hepPackage):
 
     depends_on('cepcsw')
     
-    depends_on('catch2@3.0.1:', when='+devtools')
+    depends_on('catch2@3.1:', when='+devtools')
```

Comment:
I don't think we need this. The dependency will come from podio if it's needed in podio. Also it wouldn't be possible to build an older version of podio with this change. It would be better to just remove the version and let the packages require whichever version they need.

Reply:
I just realized we need to keep `@3:` because prior to that `catch2` did not build the static library to link against.

Reply:
We still don't need `@3:` since it will try to install the latest version anyway so it would only be useful in case someone was having an environment with something that required 2.x and 3.x at the same time, or specifying manually to install catch2 but this is rare